modifiers/godot/voxel_modifier_gd.cpp

```diff
@@ -115,16 +85,55 @@ void VoxelModifier::_notification(int p_what) {
 				_modifier_id = 0;
 			}
 		} break;
+		case Node3D::NOTIFICATION_TRANSFORM_CHANGED: {
+			if (_volume == nullptr) {
+				Node *parent = get_parent();
+				VoxelLodTerrain *volume = Object::cast_to<VoxelLodTerrain>(parent);
+				// Using get_parent_node_3d when terrain is immediate parent
+				// would be a breaking change as it returns null if node is toplevel
+				if (volume == nullptr) {
+					Node3D *grandparent = get_parent_node_3d();
+					volume = Object::cast_to<VoxelLodTerrain>(grandparent);
+					while (grandparent != nullptr && volume == nullptr) {				
+						grandparent = grandparent->get_parent_node_3d();
+						volume = Object::cast_to<VoxelLodTerrain>(grandparent);
+					}
+				}
+				_volume = volume;
```

Comment:
Parenting logic should not be in transform notifications. Other code might break unexpectedly if no transformations occur.

Reply:
It seems that the PARENTING notification is fired for packed scenes before they are attached to the main scene, so the volume is never found and the modifier is not initialized.
As a workaround, I moved the parenting code to the transform update. I suspect that a transform notification will always be fired after a parenting event because nodes do not have a global transform before they exist in the tree, however, I could be wrong about this. It appears to work in my tests.

Reply:
I would still not rely on the transform notification for parenting, at least not alone, because it's not what it's for... Alternatives are to either make a special getter function that attempts to get the volume each time it is found null, or to defer the whole logic to cases where the volume enters the scene tree, but that also has annoying drawbacks such as reparenting the terrain causing total remeshing for no reason...